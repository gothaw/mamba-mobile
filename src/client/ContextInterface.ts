export interface ContextInterface {
  container: object
}