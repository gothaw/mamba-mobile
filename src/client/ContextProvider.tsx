import React, { createContext, FunctionComponent } from 'react';

import { ContextInterface } from "../types";
import { theme } from "../theme";

type Props = {
  children: React.ReactNode
};

const Context = createContext<ContextInterface | null>(null);

const ContextProvider: FunctionComponent<Props> = ({children}) => {
  return (
    <Context.Provider value={theme}>
      {children}
    </Context.Provider>
  );
};

export { Context, ContextProvider };
