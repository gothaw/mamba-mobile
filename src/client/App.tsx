import MainController from "./MainController";

const App = () => {
  return (
    <MainController/>
  );
}

export default App;