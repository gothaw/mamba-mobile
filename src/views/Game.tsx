import React, { FunctionComponent, useContext, useEffect, useRef } from "react";
import { ImageBackground, View } from "react-native";
import Canvas from "react-native-canvas";

import { Context } from "../client/ContextProvider";
import { Images } from "../config";
import BackgroundGradient from "../containers/BackgroundGradient";

interface Props {
}

const Game: FunctionComponent<Props> = () => {
  const ref = useRef(null);

  const context = useContext(Context);

  useEffect(() => {
    if (ref.current) {
      const ctx = ref.current.getContext("2d");
      ctx.globalAlpha = 0;

      if (ctx) {
        ctx.globalAlpha = 1;
        ctx.fillStyle = "rgba(255, 255, 255, 0.0)";
        ctx.fillRect(0, 0, 20, 20);
        ctx.globalAlpha = 0;
      }
    }
  }, [ref]);

  return (
    <BackgroundGradient>
      <View style={{...context.theme.viewWrapper}}>
        <View style={{...context.theme.gameWrapper}}>
          <ImageBackground
            source={Images.GameBackground}
            style={{...context.theme.gameBackgroundImg}}/>
          <Canvas
            ref={ref}
            style={{...context.theme.gameCanvas}}/>
        </View>
      </View>
    </BackgroundGradient>
  );
};

export default Game;
