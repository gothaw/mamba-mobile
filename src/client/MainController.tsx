import { createContext } from "react";
import { Route, NativeRouter, Routes } from "react-router-native";
import Game from "../views/Game";
import MainMenu from "../views/MainMenu";

const MainController = () => {
  const Context = createContext({});

  return (
    <Context.Provider value={{}}>
      <NativeRouter>
        <Routes>
          <Route path="/" element={<MainMenu/>}/>
          <Route path="/game" element={<Game/>}/>
        </Routes>
      </NativeRouter>
    </Context.Provider>
  );
}

export default MainController;