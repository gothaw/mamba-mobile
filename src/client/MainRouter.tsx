import React, { FunctionComponent, useContext } from "react";
import { View } from "react-native";
import { NativeRouter, Route, Routes } from "react-router-native";

import Icon from "../components/Icon";
import { Images, Paths } from "../config";
import Game from "../views/Game";
import MainMenu from "../views/MainMenu";

import { Context } from "./ContextProvider";

interface Props {
}

const MainRouter: FunctionComponent<Props> = () => {

  const context = useContext(Context);

  return (
    <View style={{...context.theme.container}}>
      <NativeRouter>
        <Routes>
          <Route path={Paths.MainMenu} element={<MainMenu/>}/>
          <Route path={Paths.Game} element={<Game/>}/>
        </Routes>
        <Icon
          src={Images.MenuIcon}
          style={context.theme.mainMenuIconWrapper}
          imgStyle={context.theme.mainMenuIconImg}
        />
      </NativeRouter>
    </View>
  );
};

export default MainRouter;
