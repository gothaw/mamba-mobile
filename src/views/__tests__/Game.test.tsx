import { NativeRouter, Route, Routes } from "react-router-native";
import { act, render, waitFor } from "@testing-library/react-native";

import { ContextProvider } from "../../client/ContextProvider";
import Game from "../Game";

describe("Game", () => {
  it("should render without crashing", async () => {
    const game = render(
      <ContextProvider>
        <NativeRouter>
          <Routes>
            <Route path={"/"} element={<Game/>} />
          </Routes>
        </NativeRouter>
      </ContextProvider>
    );

    await act(async () => {
      await new Promise(resolve => setImmediate(resolve));
    });

    await waitFor(() => {
      expect(game.toJSON()).toMatchSnapshot();
    });
  });
});