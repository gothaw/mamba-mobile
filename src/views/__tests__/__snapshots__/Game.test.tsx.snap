// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should render without crashing 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4290491705,
        4287202352,
      ],
      "endPoint": [
        0.25,
        1,
      ],
      "locations": undefined,
      "startPoint": [
        0,
        0,
      ],
    }
  }
  style={
    {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "position": "relative",
      }
    }
    testID="game"
  >
    <View
      style={
        {
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 600,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          {
            "borderColor": "#c9c9c9",
            "borderWidth": 4,
            "elevation": 10,
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
            "top": 0,
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "height": "100%",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          [
            {
              "flex": 1,
            },
            {
              "borderColor": "#c9c9c9",
              "borderWidth": 8,
              "flex": null,
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "bottom": "5%",
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <View
        collapsable={false}
        forwardedRef={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(0,0,0,0.4)",
            "borderRadius": 75,
            "height": 150,
            "justifyContent": "center",
            "transform": [
              {
                "translateX": 0,
              },
              {
                "translateY": 0,
              },
            ],
            "width": 150,
          }
        }
      >
        <View
          collapsable={false}
          forwardedRef={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.8)",
              "borderRadius": 37.5,
              "height": 75,
              "justifyContent": "center",
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateY": 0,
                },
              ],
              "width": 75,
            }
          }
        />
      </View>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
